pkg/logql/evaluator.go

```diff
@@ -124,20 +125,20 @@ func Sortable(q Params) (bool, error) {
 }
 
 // Evaluator is an interface for iterating over data at different nodes in the AST
-type Evaluator interface {
-	SampleEvaluator
+type Evaluator[V any] interface {
+	SampleEvaluator[V]
```

Comment:
I've made this change originally to easily switch between the return types. However, it seems that it should always be `promql_parser.Value` or rather our own interface.